dlhub_sdk/utils/publish.py

```diff
@@ -0,0 +1,244 @@
+    # index = mdf_toolbox.translate_index(idx)
+
+    ingestable = task
+    d = [convert_dict(ingestable, str)]
+
+    glist = []
+    visible_to = task['dlhub'].get('visible_to', ['public'])
+
+    # Add public so it isn't an empty list
+    if len(visible_to) == 0:
+        visible_to = ['public']
+
+    for document in d:
+        gmeta_entry = mdf_toolbox.format_gmeta(document, visible_to, iden)
+        glist.append(gmeta_entry)
+    gingest = mdf_toolbox.format_gmeta(glist)
+
+    logger.info("ingesting to search")
+    logger.info(gingest)
+
+    GLOBUS_SEARCH_WRITER_LAMBDA = "https://7v5g6s33utz4l7jx6dkxuh77mu0cqdhb.lambda-url.us-east-2.on.aws/"
+
+    # Get the authorization header token (string for the headers dict)
+    #header = self.authorizer.get_authorization_header()
+
+    http_response = requests.post(
+        GLOBUS_SEARCH_WRITER_LAMBDA,
+        headers={"Authorization": header},
+        json={'document_file': gingest})
+    if http_response.status_code != 200:
+        raise Exception(http_response.text)
+    logger.info("Ingestion of {} to DLHub servables complete".format(iden))
+
+
+def mint_identifier(task):
+    """Mint a new identifier and return it."""
+    identifiers_namespace = '1EGOGHSs9RAtq'
+    identifier = None
+    logger.debug("Creating identifier")
+    try:
+        # client = identifiers_client(config)
+        serv_location = "https://%s%s" % ("dlhub.org/servables/", task['dlhub']['id'])
+        # safe encode the location
+        serv_location = urllib.parse.quote_plus(serv_location, safe='')
+        # can't be too safe!
+        serv_location = urllib.parse.quote_plus(serv_location, safe='')
+        # Create the petrel link
+        serv_location = f'https://petreldata.net/dlhub/detail/{serv_location}'
+        visible_to = ['public']
+
+        dataset_identifier = client.create_identifier(
+            namespace=identifiers_namespace,
+            location=[serv_location],
+            metadata={
+                'uuid': task['dlhub']['id'],
+                'shorthand_name': task['dlhub']['shorthand_name']
+            },
+            landing_page=serv_location,
+            visible_to=visible_to)
+
+        res = dataset_identifier.data
+        identifier = res['identifier']
+        logger.debug(f"Created identifier: {identifier}")
+    except Exception as e:
+        logger.error(e)
+    return identifier
+
+
+def register_funcx(task, container_uuid, funcx_client):
+    """Register the function and the container with funcX.
+
+    Parameters
+    ----------
+    task : dict
+        A dict of the task to publish
+    container_uuid:  str
+    funcx_client:  the funcx client to register with
+
+    Returns
+    -------
+    str
+        The funcX function id
+    """
+
+    fxc = funcx_client
+
+    description = f"A container for the DLHub model {task['dlhub']['shorthand_name']}"
+    try:
+        description = task['datacite']['descriptions'][0]['description']
```

Comment:
so are you just overwriting the description? why not put Line 158 in the `except` clause?